src/build/env.py

```diff
@@ -1,312 +1,293 @@
 """
 Creates and manages isolated build environments.
 """
+
 import abc
-import functools
 import logging
 import os
 import platform
 import shutil
-import subprocess
 import sys
 import sysconfig
 import tempfile
 
-from types import TracebackType
-from typing import Callable, Iterable, List, Optional, Tuple, Type
-
-import build
-
+from typing import Dict, Generic, Iterable, Optional, Sequence, Tuple, TypeVar, cast, overload
 
-if sys.version_info < (3, 8):
-    import importlib_metadata as metadata
-else:
-    from importlib import metadata
+from ._helpers import cache, check_dependency, default_runner
 
-try:
-    import virtualenv
-except ModuleNotFoundError:
-    virtualenv = None
 
+_logger = logging.getLogger(__name__)
 
-_logger = logging.getLogger('build.env')
+IsolatedEnvType = TypeVar('IsolatedEnvType', bound='IsolatedEnv')  #: :class:`IsolatedEnv` type alias.
 
 
 class IsolatedEnv(metaclass=abc.ABCMeta):
-    """Abstract base of isolated build environments, as required by the build project."""
+    """ABC for isolated build environments."""
 
-    @property
     @abc.abstractmethod
-    def executable(self) -> str:
-        """The executable of the isolated build environment."""
-        raise NotImplementedError
+    def create(self, path: str) -> None:
```

Comment:
Not all implementation need to be tied to a path, this API is not very nice :confused:

Reply:
The `IsolatedEnv{Builder,Manager}` needs to pass the path to the isolated env somehow.

Reply:
Hence I think this should be tied to that specific class, which has that requirement. This can be done via a `classmethod` context manager.